recipes/sdsl-lite/all/conanfile.py

```diff
@@ -0,0 +1,59 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+import yaml
+
+
+class SDSLLite(ConanFile):
+    name = "sdsl-lite"
+    libname = "sdsl"
+    exports_sources = "CMakeLists.txt"
+    exports = "submoduledata.yml"
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    _source_subfolder = "sdsl-lite"
+    _build_subfolder = "build_folder"
+    version = "2.1.1"
+    git_url = "https://github.com/simongog/sdsl-lite.git"
+    license = "GNU GPL"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "SDSL - Succinct Data Structure Library"
+    homepage = "https://github.com/simongog/sdsl-lite"
+    topics = ("conan", "sdsl", "succint", "data structure")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+        submodule_filename = os.path.join(
+            self.recipe_folder, 'submoduledata.yml')
```

Comment:
Please don't use git submodules.
By downloading dependencies here, you're actually vendoring libraries and potentially creating duplicate libraries in a dependency tree.

Instead you should (try to) patch the cmake build script to e.g.:
- don't build the tests so you don't need google test. This can be done by commenting out [this line](https://github.com/simongog/sdsl-lite/blob/c32874cb2d8524119f25f3b501526fe692df29f4/CMakeLists.txt#L151) from the cmake script and modifying [this cmake script](https://github.com/simongog/sdsl-lite/blob/master/external/CMakeLists.txt).
- link to the libdivsufsort provided by cci instead of a vendored version (or only use its headers).

Don't hesitate to ask questions if you're stuck somewhere.